cypress/integration/advanced-functionalities/admin/system/integration-gateway/Route_filters.spec.js

```diff
@@ -0,0 +1,176 @@
+/// <reference types="cypress" />
+const adminURL = Cypress.env('ADMIN_URL')
+const email = Cypress.env('USER_EMAIL')
+const password = Cypress.env('USER_PASSWORD')
+
+describe('Test for checking the route filters of an integration gateway', () => {
+  before(() => {
+    if (!window.sessionStorage.getItem('auth.refresh-token')) {
+      cy.login({ email, password, url: adminURL })
+    }
+  })
+
+  context('Test for checking the route filters of an integration gateway', () => {
+    it('should be able to enable and disable the pre filtering header filter', () => {
+      cy.visit(adminURL + '/')
+      // We wait 3s in order the page to be fully loaded
+      cy.wait(3000)
+      cy.get('.nav-sidebar').contains('Integration Gateway').click()
+      // We wait 1s in order the page to be fully loaded
+      cy.wait(1000)
+      cy.get('[data-test-id="input-search"]').type('/testEdited')
+      cy.get('#resource-list > tbody > tr:last > td:last > a').click()
+      // We wait 1s in order the page to be fully loaded
+      cy.wait(1000)
+      cy.get('[data-test-id="card-filter-list"]').within(() => {
+        cy.get('[data-test-id="dropdown-add-filter"] > button:eq(0)').click()
+        cy.get('[data-test-id="header"]:eq(0)').click()
+      })
+      // We click on Save & Close
+      cy.get('.modal-footer > .btn-primary').click()
+      cy.get('[data-test-id="card-filter-list"]').within(() => {
+        // We check that the filter is added and the status Enabled exists
+        cy.contains('Enabled').should('exist')
+        // Next we remove the filter and see if it gets removed
+        cy.contains('Remove').click()
+        cy.get('[data-test-id="no-filters"]').should('exist')
+      })
+    })
+
+    it('should be able to enable and disable the pre filtering profiler filter', () => {
+      cy.get('[data-test-id="card-filter-list"]').within(() => {
+        cy.get('[data-test-id="dropdown-add-filter"] > button:eq(0)').click()
+        cy.get('[data-test-id="profiler"]:eq(0)').click()
+      })
+      // We click on Save & Close
+      cy.get('.modal-footer > .btn-primary').click()
+      cy.get('[data-test-id="card-filter-list"]').within(() => {
+        // We check that the filter is added and the status Enabled exists
+        cy.contains('Enabled').should('exist')
+        // Next we remove the filter and see if it gets removed
+        cy.contains('Remove').click()
+        cy.get('[data-test-id="no-filters"]').should('exist')
+      })
+    })
+
+    it('should be able to enable and disable the pre filtering query parameters filter', () => {
+      cy.get('[data-test-id="card-filter-list"]').within(() => {
+        cy.get('[data-test-id="dropdown-add-filter"] > button:eq(0)').click()
```

Comment:
After selecting an option it's color is changed to gray. Can you, please, test it as well?